Feature/rafael sapalo/#87/add the token consitency service

Rafael-Sapalo opened the pull request.

Rafael-Sapalo committed:
feat: add refresh token generation utility to auth service JWT utils

- Added REFRESH_TOKEN_EXP constant set to 30 days (2,592,000 seconds)
- Created generateRefreshToken function using HS256 algorithm with REFRESH_SECRET_KEY
- Added environment variable validation for REFRESH_SECRET_KEY
- Implemented error handling for refresh token generation failures
- Set refresh token expiration and issued at claims in JWT payload

Rafael-Sapalo committed:
refactor: convert JWT utilities to class-based architecture and add login validation

- Refactored JWT utility functions into JwtUtils static class for better organization
- Changed tokenPayload type to PascalCase TokenPayload for naming consistency
- Converted ACCESS_TOKEN_EXP and REFRESH_TOKEN_EXP to private static readonly class constants
- Converted generateAccessToken, generateRefreshToken, verifyAccessToken to static methods
- Added verifyRefreshToken static method with REFRESH_SECRET_KEY validation

Rafael-Sapalo committed:
feat: add email field to AuthRepository queries and create TokenRepository for refresh token management

- Added email field to AuthUserCredentials type definition
- Updated getUserById and getUserByEmail methods to include email in select queries
- Created TokenRepository class with createToken method for storing refresh tokens
- Added device field support for token creation to track user sessions

Rafael-Sapalo committed:
feat: add device field and default timestamps to token table schema

- Added device text field as required column to track user sessions
- Added defaultNow() to created_at timestamp for automatic timestamp on insert
- Added defaultNow() to updated_at timestamp for automatic timestamp on insert and update

Rafael-Sapalo committed:
feat: implement login endpoint with JWT authentication and refresh token support

- Added loginValidation schema import for request validation
- Added TokenRepository instance to AuthController for refresh token management
- Implemented login handler with email/password validation using sValidator middleware
- Added password verification using Bun.password.verify against hashed database password
- Added parallel generation of access and refresh tokens using Promise.all
- Added refresh token storage

Rafael-Sapalo committed:
feat: add secure HTTP-only cookie for refresh token storage in login endpoint

- Added setCookie import from hono/cookie to AuthController
- Configured refresh token cookie with httpOnly, secure, and sameSite flags for security
- Set cookie path to /auth/refresh and maxAge to REFRESH_TOKEN_EXP (30 days)
- Changed REFRESH_TOKEN_EXP visibility from private to public in JwtUtils for cookie configuration access

Rafael-Sapalo committed:
feat: implement refresh token endpoint with token rotation and session management

- Added getCookie import to AuthController for reading refresh token from cookies
- Implemented refresh token handler with token validation and verification
- Added getTokenByToken method to TokenRepository to retrieve session by token hash
- Added updateToken method to TokenRepository for token rotation with session ID tracking
- Added deleteToken method to TokenRepository for session cleanup
- Added error handling for missing

Mord0rak commented on elyrii_server/services/auth/src/controller/auth.controller.ts:
```diff
@@ -60,7 +101,51 @@ class AuthController {
+
+        let payload;
+        try {
+            payload = await JwtUtils.verifyRefreshToken(oldRefreshToken);
```
il y a t'il toujours une utilite au ligne 154-156 du coup ?

Rafael-Sapalo replied:
Non elle peut être supprimé

Rafael-Sapalo committed:
feat: remove unused verifyToken endpoint from AuthController

- Removed verifyToken handler and its TODO comment from AuthController
- Cleaned up unused token verification endpoint that was not implemented
